### PR TITLE
[SecurityBundle] fix LDAP-based HTTP Basic Auth entry point registration

### LINKED ISSUE #39374: LDAP complains about missing method
**Symfony version(s) affected**: 5.2.0

**Description**  
After upgrading from 5.1 to 5.2.0 the error `Call to undefined method Symfony\Bundle\SecurityBundle\DependencyInjection\Security\Factory\HttpBasicLdapFactory::registerEntryPoint() in /app/vendor/symfony/security-bundle/DependencyInjection/Security/Factory/HttpBasicLdapFactory.php:48` occurs.

**How to reproduce**  
Upgrade from 5.1 to 5.2 with LDAP Configuration.
security.yaml:

```yaml
security:
  providers:
    admin_provider:
      entity:
        class: App\Entity\Admin\User
        property: username
    user_provider:
      id: App\Security\UserProvider
    my_ldap:
      ldap:
        service: Symfony\Component\Ldap\Ldap
        base_dn: 'OU=Foo,dc=bar,dc=example,dc=com'
        search_dn: 'CN=user,OU=Foo,DC=example,DC=example,DC=com'
        default_roles: ROLE_ADMIN
        search_password: '%env(LDAP_PASSWORD)%'
        uid_key: 'sAMAccountName'
  firewalls:
    dev:
      pattern: ^/(_(profiler|wdt|fragment)|css|images|js)/
      security: false
    admin:
      provider: my_ldap
      anonymous: true
      lazy: true
      pattern:    ^/admin
      http_basic_ldap:
        service: Symfony\Component\Ldap\Ldap
        dn_string: 'EXAMPLE_DOMAIN\{username}'
        provider: my_ldap
      logout:
        path:   /logout
        target: /admin
    main:
      provider: user_provider
      anonymous: true
      lazy: true
      pattern:    ^/
      form_login:
        login_path: _homepage
        check_path: login
        provider: user_provider
        username_parameter: login_form[_username]
        password_parameter: login_form[_password]
      logout:
        path:   /logout
      guard:
        authenticators:
          - App\Security\IpAuthenticator
        provider: user_provider
        entry_point: App\Security\IpAuthenticator
[...]

```

**Additional context**  

grafik

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 5.2
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #39374
| License       | MIT
| Doc PR        |

### CODE REVIEW

**Comment by xabbuh** [comment]
this change was missed in #39153 (cab0672248f95a01b578eb69cc5ce712dcf6a49f)

**Review by chalasr — APPROVED** [review]

File: src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/HttpBasicLdapFactory.php [hunk 1]
```diff
@@ -45,7 +45,11 @@ public function create(ContainerBuilder $container, string $id, array $config, s
         ;
 
         // entry point
-        $entryPointId = $this->registerEntryPoint($container, $id, $config, $defaultEntryPoint);
+        $entryPointId = 'security.authentication.basic_entry_point.'.$id;
```
Comment (jderusse) [inline src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/HttpBasicLdapFactory.php]: In `HttpBasicFactory` the code is different:
```php
// entry point
$entryPointId = $defaultEntryPoint;
if (null === $entryPointId) {
  $entryPointId = 'security.authentication.basic_entry_point.'.$id;
  $container
    ->setDefinition($entryPointId, new ChildDefinition('security.authentication.basic_entry_point'))
    ->addArgument($config['realm'])
  ;
}
```

Comment (xabbuh) [reply src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/HttpBasicLdapFactory.php]: good catch, fixed

**Review by jderusse — APPROVED** [review]

**Comment by wouterj** [comment]
Thank you Christian! :rocket:

**Pull request merged**